app/services/foreman_rh_cloud/cloud_connector.rb

```diff
@@ -12,11 +12,20 @@ def install
       target_host = foreman_host
       composer = nil
 
+      input = {
+        :satellite_cloud_connector_user => service_user.login,
+        :satellite_cloud_connector_password => token_value,
+      }
+
+      if (http_proxy = Setting[:http_proxy])
+        input[:satellite_cloud_connector_http_proxy] = ForemanRhCloud.proxy_setting(logger: Foreman::Logging.logger('app'))
```

Comment:
```suggestion
      if (http_proxy = ForemanRhCloud.proxy_setting(logger: Foreman::Logging.logger('app')))
        input[:satellite_cloud_connector_http_proxy] = http_proxy
```

We should use the same data source for both the condition and the assignment. Otherwise you will have edge cases where the condition would be falsy, but the proxy would be required.